### ISSUE TITLE
Permissions array operators: Check if array column contains value

### ISSUE THREAD

**Issue by stonymahony** [hasura/graphql-engine#7346]
I am currently testing Hasura for use in a productive system and am quite impressed, good work!

Unfortunately, I have not yet been able to realise an auhtorisation use case, the following scenario:

_A customer has several divisions, users can belong to several divisions and have several different roles there.
To map this, there is a table userDivisionRoles in the database with the columns userId, divisionId and roles (an array of the enum DivisionRole)._

I would like to check for incoming requests in the table userDivisionRoles:
- userId = X-Hasura-User-Id
- divisionId = X-Hasura Division-Id
- roles: Contains X-Hasura-Role / X-Hasura-Default-Role (= user has the appropriate role in the division).

So basically what is needed is an array operator "contains" that checks whether a certain value (right side) is contained in a column of type array (left side).

Does such an operator exist and have I overlooked it, or is this simply not possible so far?
How could this be modelled instead? 

Thank you very much!

**Comment by mylesfarm** [comment]
@stonymahony I'm also having trouble with this use-case at the moment. I'm passing an array of "allowed org" uuids in my JWT claim using Postgres array literal format like so:

```json
{
  "https://hasura.io/jwt/claims": {
    "x-hasura-default-role": "user",
    "x-hasura-allowed-roles": [
      "user"
    ],
    "x-hasura-allowed-orgs": "{24b624b5-e85b-4fab-a93e-0c0e4def9792,8d47443c-4a22-454d-ba3e-cb0011264919}",
    "x-hasura-user-id": "sms|610b3bf40858457944c10000"
  },
  "iss": "https://server.us.auth0.com/",
  "sub": "sms|610b3bf40858457944c10000",
  "aud": [
    "https://hasura.io/jwt/claims",
    "https://server.us.auth0.com/userinfo"
  ],
  "iat": 0000000000,
  "exp": 0000000000,
  "azp": "1QBxxqgy9aYvE5YxOVllT0krTTJvyTM4",
  "scope": "openid profile email",
  "gty": "password"
}
```

I then have a permission on my orgs table with custom check `{"id":{"_in":["X-Hasura-Allowed-Orgs"]}}`, per Hasura docs.

However, when I run this query, using an org id contained in the claim:

```graphql
query ($orgId: uuid!) {
  orgs_by_pk(id: $orgId) {
    id
    name
  }
}
```
Variables:

```json
{
  "orgId": "24b624b5-e85b-4fab-a93e-0c0e4def9792"
}
```

I get the following error:

```json
{
  "errors": [
    {
      "extensions": {
        "path": "$",
        "code": "data-exception"
      },
      "message": "invalid input syntax for type uuid: \"X-Hasura-Allowed-Orgs\""
    }
  ]
}
```

Any help from the Hasura team would be much appreciated! Thank you.

**Comment by mylesfarm** [comment]
**Edit:** I figured it out and it's working fine now. When I set up the permissions, my `X-Hasura-Allowed-Orgs` session variable array was _inside_ of the `_in` array, when it should have been the value of the array like this:

![Screen Shot 2022-01-14 at 11 32 16](https://user-images.githubusercontent.com/46573464/149574362-3173f0e4-bed9-47b5-ba97-924b7f32fd46.png)

**Comment by 95Kevid** [comment]
Having the same issue. We need a way to test that a related table array does not 'contain' a certain value. For example, 
dataset does not contain a 'limitation' from an array in a related table. The _in and _nin operators does not fit in the scenario. Would be nice to have a _contains operator for this.
